### ISSUE TITLE
SDV 1.4 and SMAPI 3.0 compatibility

### ISSUE THREAD

**Issue by purrplingcat** [purrplingcat/PurrplingMod#35]
Compatibilize mod with SMAPI 3.0/SDV 1.4

From discussion with @Pathoschild:

Trace output:
```
NpcAdventure (from Mods\NPC Adventures\NpcAdventure.dll)...
   Broken code in NpcAdventure.dll: reference to StardewModdingAPI.IMonitor.ExitGameImmediately (no such method).
   Broken code in NpcAdventure.dll: reference to StardewValley.GameLocation.playSound (no such method).
   Broken code in NpcAdventure.dll: reference to StardewValley.Game1.createItemDebris (no such method).
   Detected filesystem access (System.IO.File type) in assembly NpcAdventure.dll.
```

There are two main issues: 
1. IMonitor.ExitGameImmediately no longer exists in SMAPI 3.0. Ideally you shouldn't force a game crash, but see below if you really need to.
2. Some of the game methods have changed. Those will probably just need a quick recompile, and possibly some minor changes.

Here's how to force a game crash without ExitGameImmediately:
```
this.Monitor.Log("Forcing game crash: something went catastrophically wrong", LogLevel.Error);
void InfiniteRecursiveLoop() => InfiniteRecursiveLoop();
InfiniteRecursiveLoop();
```
**GameLocation.playSound issue**
The method has a new optional argument. You just need to recompile the mod, no changes needed.
There's a lot of changes like that, which are covered by this line in the migration guide:

Various method signatures have changed [...]. In most cases just recompiling will fix those. (Don't forget to update the manifest.json version!)

**Comment by purrplingcat** [comment]
Proposed issue automatically, because SDV 1.4 not out to public. Make this issue when SDV 1.4 and SMAPI 3.0 will be released

**Comment by purrplingcat** [comment]
Needs revision. At **26th November** will be released new SDV version 1.4 and probably new SMAPI. After new SDV was out, check discompatabilities and compatibilize it. This ticket will be blocker after SDV and SMAPI release.

**Comment by purrplingcat** [comment]
DONE. Released 0.4.0.
